### PR TITLE
commit with stretch goals

### PR DESCRIPTION
**Pull request by sprainhill**

### CODE REVIEW

File: src/components/Counter.js [hunk 1]
```diff
@@ -1,42 +1,50 @@
...
-    incrementAsync = () => {
-        // Stretch Problem: Implement an increment function that
-        // increments after waiting for one second
-    };
+  incrementAsync = () => {
+    // Stretch Problem: Implement an increment function that
+    // increments after waiting for one second
+    setTimeout(() => {
+      this.props.increment();
+    }, 1000);
```
Comment (Fractured2K) [inline src/components/Counter.js:19]: Great work on both stretch goals.

File: src/reducers/index.js [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
+        count: state.count + 1
+      };
+      break;
     case DECREMENT:
-    // Fill in the body of this case
+      return {
+        ...state,
+        count: state.count - 1
+      };
+      break;
```
Comment (Fractured2K) [inline src/reducers/index.js:26]: Perfect! 

So yesterday when we were talking you noticed `++state.count` and you said tried `state.count++`, but got some weird functionality. The reason why this happens is because of the order in which JavaScripts reads.

_e.g_
```
// Increment
let a = 1;
console.log(a++);    // 1
console.log(a);      // 2

// Decrement
let b = 1;
console.log(b--);    // 1
console.log(b);      // 0
```
When you use the increment/decrement operator after the operand, the value will be returned before the operand is increased/decreased.

```
// Increment
let a = 1;
console.log(++a);    // 2
console.log(a);      // 2

// Decrement
let b = 1;
console.log(--b);    // 0
console.log(b);      // 0
```
If you’d rather make the variable increment/decrement before returning, you simply have to use the increment/decrement operator before the operand like above

So in conclusion both ways work and they both update the value, but if the operand comes after the value the original value will be returned instead of the updated one unless called on again.

**Review by Fractured2K — COMMENTED** [review]
Great work here Rory! Keep it up!

File: src/reducers/index.js (hunk 2 above)
Comment (sprainhill) [reply src/reducers/index.js:26]: Awesome explanation, thanks!
